data_model/src/transaction.rs

```diff
@@ -172,6 +172,7 @@ pub mod model {
     #[ffi_type]
     pub struct RejectedTransaction {
         pub transaction: VersionedSignedTransaction,
+        pub block_order_index: u64,
```

Comment:
position of a transaction within a block should not be a property of a transaction, but a property of the block. You should explore way to move `block_order_index` into the block

Reply:
on 2nd thought, this might be fine. I want to see other opinions

Reply:
Here is my suggestion. Currently we have block defined like this:
```rs
        pub struct CommittedBlock {
            pub header: BlockHeader,
            pub rejected_transactions: Vec<RejectedTransaction>,
            pub transactions: Vec<VersionedSignedTransaction>,
        }
```

we could define it like this:
```rs
        pub struct CommittedBlock {
            pub header: BlockHeader,
            /// This Vec contains all transactions rejected and accepted
            pub transactions: Vec<VersionedSignedTransaction>,
            /// This Vec contains location in the previous Vec + rejection reason
            pub rejections: Vec<(u64, TransactionRejectionReason)>
        }
```

Reply:
I will take your suggestion.

Reply:
Actually doing this change would require a lot more refactoring. Since I am not do do such thing I think we should merge this PR as is. Roman and I have already discussed how this could be unified into a single transaction queue with a boolean to indicate transaction failure. I have created the issue here. #3623

Reply:
> on 2nd thought, this might be fine. I want to see other opinions

I also think that order information should be part of the block.
It will code cleaner i think. 
Additionally i don't think that we should give this information to user on transactions.

Reply:
Ok I will pursue the single tx array.

Reply:
Why can't we just store an array of marked transactions? Why to store rejected transactions in a separate field. I can't see where we use this

Reply:
Only place i can think of is merkle tree root hash, but we can compute this hash anyway.